lxd/instance/drivers/driver_lxc.go

```diff
@@ -2037,23 +2062,30 @@ func (d *lxc) startCommon() (string, []func() error, error) {
 				}
 			}
 
-			if d.state.OS.Shiftfs && !d.IsPrivileged() && diskIdmap == nil {
-				// Host side mark mount.
-				err = lxcSetConfigItem(d.c, "lxc.hook.pre-start", fmt.Sprintf("/bin/mount -t shiftfs -o mark,passthrough=3 %s %s", strconv.Quote(d.RootfsPath()), strconv.Quote(d.RootfsPath())))
-				if err != nil {
-					return "", nil, errors.Wrapf(err, "Failed to setup device mount shiftfs '%s'", dev.Name)
-				}
+			if !d.IsPrivileged() && diskIdmap == nil {
+				if SupportsIdmappedMounts(storageType, d.RootfsPath()) {
```

Comment:
@brauner can you re-use the `idmappedMounts` var here rather than call `SupportsIdmappedMounts` again?

Reply:
I'm not sure I quite understand. This would require me to open code the locking around the map.